### ISSUE TITLE
Terra Thread Posting Bug

### ISSUE THREAD

**Issue by alexyoung23j** [hicommonwealth/commonwealth#3245]
Terra user is unable to post a thread in Terra community. See screenshots here: https://www.notion.so/buildcommon/Terra-user-unable-to-post-thread-38b422bb6f42400ab01bc6af11bd8484?pvs=4

Possibly related to cloudflare config, Jake is investigating.

**Comment by alexyoung23j** [comment]
Jake did not notice anything obvious related to cloudflare, need to investigate further.

**Comment by alexyoung23j** [comment]
Ian and I were unable to reproduce. Here is a walkthrough of how we approached this:

**Hypothesis 1**: Thread posting is not working because there is something about the content the user posted (i.e. length, images) that was not working.  

**Experiment 1**: Copied the exact post contents that the terra user wanted to post (obtained from Mark in bd) and was able to create post normally. Hypothesis rejected. 

**Hypothesis 2**: The frontend token balance checking is broken, and users who do not actually have sufficient token balances to post are being given the option to post (button is not disabled, etc).

**Experiment 2**: First, removed the token gating from the topic to check if posting would work as expected. It did. Then, in the place of the frontend token balance cache check, inserted a hardcoded token balance value greater than the token threshold. Posting ability confirmed. Then made the token balance smaller than the threshold. Posting inability confirmed (the desired behavior). Hypothesis rejected. 

**Hypothesis 3**: Backend token balance check is broken; users who are correctly passing token balance checks on frontend are failing them on backend. This is preventing the post from being created.

**Experiment 3**: Repeat experiment 2 on the backend. Hardcoded the backend token check value in the same manner. Posting worked as expected. Hypothesis rejected. 

**Hypothesis 4**: User was an admin, so the UI was showing them topics they should not have the ability to post to without meeting token threshold, and they were able to attempt a post as a result.

**Experiment 4**: User was not admin (confirmed with BD). And tried as a local admin and as a local non admin, behavior was as expected. Hypothesis rejected. 

**Hypothesis 5**: Token balance cache itself is broken in some idiosyncratic way related to cosmos.

**Experiment 5**: Would expect other token balances to be broken and this report came exclusively from terra users. Moreover, in the previous experiments we simulated the situation where the TBC was either providing a false positive or a false negative on the threshold check, and verified that the UI enabled/disabled the post thread button correctly. And verified that the error that would be thrown on a misconfigured TBH was a different error than the one posted by the bug reporter. But did not dig into TBC code directly. Hypothesis rejected (tepidly?).
